src/hyp_map.rs

```diff
@@ -96,8 +96,44 @@ impl HypMapFixedRegion {
     }
 }
 
-/// Represents a virtual address region that must be allocated and populated to be mapped.
-struct HypMapPopulatedRegion {
+// Represents an area that must be reset to restore U-mode original state.
+struct UmodeResetArea {
+    // The physical address where this region starts.
+    paddr: PageAddr<SupervisorPhys>,
+    // The size of the memory mapped for this region. A PageSize::Size4k multiple, bigger than `size`.
+    mapped_size: u64,
+    // The size in bytes of this region.
+    size: usize,
+    // Data to be populated at the beginning of the area.
+    data: Option<&'static [u8]>,
+}
+
+impl UmodeResetArea {
+    fn reset(&self) {
```

Comment:
If you are resetting memory, do you have to flush the cache too?

Reply:
Risc-V caches are a bit of a cloudy area, at least to me. Are you implying some form of virtual address tagging in the cache?
If this is for RVWMO (thanks @abrestic-rivos) then I shouldn't have to because this memory is strictly per-cpu.